### PR TITLE
Blazor Components Lifecycle - OnParametersSet

### PR DESCRIPTION
**Pull request by hakenr**
`OnParametersSet[Async]` is not "executed each time the component is rendered". If you just handle an event inside the component (such as button onclick), the component gets re-rendered without calling the `OnParametersSet[Async]` (`SetParametersAsync`).
Try this sample page:
```cshtml
@page "/handleevent"

<div>
	<button @onclick="HandleClick">Handle event</button>
	@something
</div>

@code
{
	private string something = String.Empty;

	private void HandleClick()
	{
		something = "something";
	}
}
```

### CODE REVIEW

**Comment by guardrex** [comment]
@danroth27 You supplied this language 👉 https://github.com/aspnet/AspNetCore.Docs/pull/10934#discussion_r256695121 ... and @rynowak didn't say anything on the later review for 3.0 GA.

@hakenr your replacement text is at our limit for localization of the docs (25 words). I'm going to break it into a list of two items on the next commit.

**Comment by guardrex** [comment]
**UPDATE**

The next round of content changes on https://github.com/aspnet/AspNetCore.Docs/pull/15630 states that ...

> These methods are executed after component initialization and each time the parent component is rendered:

... the word "parent" is added per https://github.com/aspnet/AspNetCore.Docs/issues/12896.

@danroth27 If that's going to be the correct fix there, then we can close this as a duplicate of the issue/fix.

**Comment by guardrex** [comment]
Ping @danroth27 ...

Note that you supplied the original language 👉 https://github.com/aspnet/AspNetCore.Docs/pull/10934#discussion_r256695121 ... and @rynowak didn't say anything on the later review for 3.0 GA.

**Comment by danroth27** [comment]
Since I've goofed this one before, I'm recommend we have @SteveSandersonMS review this one.

File: aspnetcore/blazor/components.md [hunk 1]
```diff
@@ -1008,7 +1008,10 @@ protected override void OnInitialized()
 }
 ```
 
-`OnParametersSetAsync` and `OnParametersSet` are called when a component has received parameters from its parent and the values are assigned to properties. These methods are executed after component initialization and each time the parent component is rendered:
+`OnParametersSetAsync` and `OnParametersSet` are called:
+
+* After component initialization.
+* When a component has received parameters from its parent and the values are assigned to properties.
```
Comment (danroth27) [inline aspnetcore/blazor/components.md]:
```suggestion
* When a component has received changed parameters from its parent and the values are assigned to properties.
```

**Comment by guardrex** [comment]
We moved this whole section of content into a new *Lifecycle* topic. The changes on this PR have been moved over to https://github.com/aspnet/AspNetCore.Docs/pull/15956. Thanks again @hakenr! :rocket:

**Pull request closed without merging**